rasa/core/training/story_reader/yaml_story_reader.py

```diff
@@ -0,0 +1,245 @@
+import logging
+from typing import Dict, Text, List, Any
+
+import rasa.utils.io
+from rasa.constants import DOCS_URL_STORIES
+from rasa.core.constants import INTENT_MESSAGE_PREFIX
+from rasa.core.events import UserUttered, SlotSet
+from rasa.core.training.story_reader.story_reader import StoryReader
+from rasa.core.training.structures import StoryStep
+from rasa.data import YAML_FILE_EXTENSIONS
+from rasa.utils.common import raise_warning
+
+logger = logging.getLogger(__name__)
+
+KEY_STORIES = "stories"
+KEY_PLAIN_STORY_NAME = "story"
+KEY_PLAIN_STORY_STEPS = "steps"
+KEY_PLAIN_STORY_USER_UTTERANCE = "user"
+KEY_SLOT_NAME = "slot"
+KEY_SLOT_VALUE = "value"
+KEY_ACTION = "action"
+KEY_PLAIN_STORY_ENTITIES = "entities"
+KEY_CHECKPOINT = "checkpoint"
+KEY_CHECKPOINT_SLOTS = "slots"
+KEY_METADATA = "metadata"
+KEY_OR = "or"
+
+
+class YAMLStoryReader(StoryReader):
+    """Class that reads the core training data in a YAML format
+
+    """
+
+    async def read_from_file(self, filename: Text) -> List[StoryStep]:
+        try:
+            yaml_content = rasa.utils.io.read_yaml_file(filename)
+
+            stories = yaml_content.get(KEY_STORIES)  # pytype: disable=attribute-error
```

Comment:
why do we need to ignore the attribute-error?

Reply:
Pytype is not very good with the union afaik. It treats `Union[List[Any], Dict[Text, Any]]` as a List only.
But thanks, I forgot to check if the `yaml_content` is a dict!